planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/util.cpp

```diff
@@ -138,6 +139,122 @@ lanelet::ConstLanelets generateBetweenEgoAndExpandedPullOverLanes(
                        -outer_road_offset);
 }
 
+Polygon2d generateObjectExtractionPolygon(
+  const lanelet::ConstLanelets & pull_over_lanes, const bool left_side, const double outer_offset,
+  const double inner_offset)
+{
+  // generate base boundary poses without orientation
+  std::vector<Pose> base_boundary_poses{};
+  for (const auto & lane : pull_over_lanes) {
+    const auto & bound = left_side ? lane.leftBound() : lane.rightBound();
+    for (const auto & p : bound) {
+      Pose pose{};
+      pose.position = createPoint(p.x(), p.y(), p.z());
+      if (std::any_of(base_boundary_poses.begin(), base_boundary_poses.end(), [&](const auto & p) {
+            return calcDistance2d(p.position, pose.position) < 0.1;
+          })) {
+        continue;
+      }
+      base_boundary_poses.push_back(pose);
+    }
+  }
+  if (base_boundary_poses.size() < 2) {
+    return Polygon2d{};
+  }
+
+  // set orientation to next point
+  for (size_t i = 0; i < base_boundary_poses.size() - 1; ++i) {
+    const auto & p = base_boundary_poses[i].position;
+    const auto & next_p = base_boundary_poses[i + 1].position;
+    const double yaw = autoware::universe_utils::calcAzimuthAngle(p, next_p);
+    base_boundary_poses[i].orientation = autoware::universe_utils::createQuaternionFromYaw(yaw);
+  }
```

Comment:
```suggestion
for (auto it = base_boundary_poses.begin(); it != std::prev(base_boundary_poses.end()); ++it) {
    const auto &p = it->position;
    const auto &next_p = std::next(it)->position;
    const double yaw = autoware::universe_utils::calcAzimuthAngle(p, next_p);
    it->orientation = autoware::universe_utils::createQuaternionFromYaw(yaw);
}
```
Maybe this is more "modern" c++?

Reply:
thanks!